From fdcade9cc58a96690706d39f03c72d252040af7e Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 01 May 2019 21:16:14 +0300
Subject: [PATCH] [SW-0] preboot: rename sources per device

<dev>/<dev>.c is actually a common code for all platforms i.e. asic,
onix, and pldm

Change-Id: Ia2736184dd6a17991271a8e2109b641cebce1525
---

diff --git a/gaudi/common.c b/gaudi/common.c
new file mode 100644
index 0000000..6fbbfa3
--- /dev/null
+++ b/gaudi/common.c
@@ -0,0 +1,191 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include <psoc_global_conf_regs.h>
+#include <pcie_aux_regs.h>
+
+#include <preboot.h>
+#include <pll.h>
+#include <uart.h>
+
+#include "common.h"
+
+void preboot_pcie_wait_for_link(void)
+{
+	while (1) {
+		unsigned int smlh, rdlh;
+
+		smlh = regmap_readl(mmPCIE_AUX_SMLH_LINK_UP);
+		rdlh = regmap_readl(mmPCIE_AUX_RDLH_LINK_UP);
+		if (smlh && rdlh)
+			break;
+	}
+}
+
+void preboot_delay(void) {}
+void preboot_putc_virt_uart(const char c) {}
+
+u64 get_cs_pad_sel(void)
+{
+	return mmPSOC_GLOBAL_CONF_PAD_SEL_53;
+}
+
+u64 get_cpu_boot_status_reg(void)
+{
+	return mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
+}
+
+u64 get_upd_pending_sts_reg(void)
+{
+	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
+}
+
+u32 hl_skip_pcie_init(void)
+{
+	/* Use DumpSel[11] to skip PCIe initialization */
+	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19);
+}
+
+u64 hl_pll_get_cpu_pll_base_off(void)
+{
+	return mmPSOC_CPU_PLL_BASE;
+}
+
+u64 hl_pll_get_cpu_cycles_per_usec(void)
+{
+	return 1200000000ULL / USEC_PER_SEC;
+}
+
+static struct hl_pll_cfg gaudi_pll_cfg[] = {
+		{
+			.name = "CPU", .base = mmPSOC_CPU_PLL_BASE,
+	//		.nr = 0, .nf = 47, .od = 1, .nb = 23,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 47, -1 }
+		}, {
+			.name = "DMA_W", .base = mmDMA_W_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "IF_W", .base = mmIF_W_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "IF_E", .base = mmIF_E_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "DMA_E", .base = mmDMA_E_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "MESH_W", .base = mmMESH_W_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "SRAM_W", .base = mmSRAM_W_PLL_BASE,
+	//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, -1, -1, -1 }
+		}, {
+			.name = "MESH_E", .base = mmMESH_E_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "SRAM_E", .base = mmSRAM_E_PLL_BASE,
+	//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, -1, -1, -1 }
+		}, {
+			.name = "MME", .base = mmPSOC_MME_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
+	//		.nr = 0, .nf = 37, .od = 1, .nb = 18,
+			.nr = 4, .nf = 331, .od = 3, .nb = 165, //DAY 1 (830MHz)
+			.div_factor = { 0, 0, 4, 9 }
+		}, {
+			.name = "TPC", .base = mmPSOC_TPC_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "HBM", .base = mmPSOC_HBM_PLL_BASE,
+			.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "NIC0", .base = mmNIC0_PLL_BASE,
+			.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
+			.div_factor = { 0, 4, 8, -1 }
+		}, {
+			.name = "NIC1", .base = mmNIC1_PLL_BASE,
+			.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
+			.div_factor = { 0, 4, 8, -1 }
+		}, {
+			.name = "NIC2_HBM", .base = mmNIC2_HBM_PLL_BASE,
+			.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "NIC2_MME", .base = mmNIC2_MME_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "NIC2_TPC", .base = mmNIC2_TPC_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		},
+};
+
+int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
+{
+	int pll_num = ARRAY_SIZE(gaudi_pll_cfg);
+	int boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
+
+	if (boot_strap & SKIP_PLL_INIT)
+		return 0;
+
+	*hl_pll_cfg = gaudi_pll_cfg;
+
+	return pll_num;
+}
+
+void hl_uart_init(void)
+{
+	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+
+	preboot_uart_init(pcie_div2_freq / HL_UART_BAUDRATE);
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+
+	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
+
+	return factor > 2 ? factor : 2;
+}
+
+int hl_pll_partial_init_get(void)
+{
+	return regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
+}
+
+void hl_pll_partial_init_set(void)
+{
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
+}
diff --git a/gaudi/common.h b/gaudi/common.h
new file mode 100644
index 0000000..0874123
--- /dev/null
+++ b/gaudi/common.h
@@ -0,0 +1,28 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+/* Use DumpSel[9] to skip PLL initialization */
+#define SKIP_PLL_INIT BIT(17)
+
+#define mmDMA_W_PLL_BASE		0x487000
+#define mmIF_W_PLL_BASE			0x488000
+#define mmIF_E_PLL_BASE			0x4A7000
+#define mmDMA_E_PLL_BASE		0x4A8000
+#define mmMESH_W_PLL_BASE		0x4C7000
+#define mmSRAM_W_PLL_BASE		0x4C8000
+#define mmMESH_E_PLL_BASE		0x4E7000
+#define mmSRAM_E_PLL_BASE		0x4E8000
+#define mmPSOC_CPU_PLL_BASE		0xC70000
+#define mmPSOC_MME_PLL_BASE		0xC71000
+#define mmPSOC_PCI_PLL_BASE		0xC72000
+#define mmPSOC_TPC_PLL_BASE		0xC73000
+#define mmPSOC_HBM_PLL_BASE		0xC74000
+#define mmNIC0_PLL_BASE			0xCF9000
+#define mmNIC1_PLL_BASE			0xD39000
+#define mmNIC2_HBM_PLL_BASE		0xD78000
+#define mmNIC2_MME_PLL_BASE		0xD79000
+#define mmNIC2_TPC_PLL_BASE		0xD7A000
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
deleted file mode 100644
index 6bfa80b..0000000
--- a/gaudi/gaudi.c
+++ /dev/null
@@ -1,191 +0,0 @@
-/*
- * (C) Copyright 2019 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-#include <psoc_global_conf_regs.h>
-#include <pcie_aux_regs.h>
-
-#include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-
-#include "gaudi.h"
-
-void preboot_pcie_wait_for_link(void)
-{
-	while (1) {
-		unsigned int smlh, rdlh;
-
-		smlh = regmap_readl(mmPCIE_AUX_SMLH_LINK_UP);
-		rdlh = regmap_readl(mmPCIE_AUX_RDLH_LINK_UP);
-		if (smlh && rdlh)
-			break;
-	}
-}
-
-void preboot_delay(void) {}
-void preboot_putc_virt_uart(const char c) {}
-
-u64 get_cs_pad_sel(void)
-{
-	return mmPSOC_GLOBAL_CONF_PAD_SEL_53;
-}
-
-u64 get_cpu_boot_status_reg(void)
-{
-	return mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
-}
-
-u64 get_upd_pending_sts_reg(void)
-{
-	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
-}
-
-u32 hl_skip_pcie_init(void)
-{
-	/* Use DumpSel[11] to skip PCIe initialization */
-	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19);
-}
-
-u64 hl_pll_get_cpu_pll_base_off(void)
-{
-	return mmPSOC_CPU_PLL_BASE;
-}
-
-u64 hl_pll_get_cpu_cycles_per_usec(void)
-{
-	return 1200000000ULL / USEC_PER_SEC;
-}
-
-static struct hl_pll_cfg gaudi_pll_cfg[] = {
-		{
-			.name = "CPU", .base = mmPSOC_CPU_PLL_BASE,
-	//		.nr = 0, .nf = 47, .od = 1, .nb = 23,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 47, -1 }
-		}, {
-			.name = "DMA_W", .base = mmDMA_W_PLL_BASE,
-	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, -1, -1 }
-		}, {
-			.name = "IF_W", .base = mmIF_W_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "IF_E", .base = mmIF_E_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "DMA_E", .base = mmDMA_E_PLL_BASE,
-	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, -1, -1 }
-		}, {
-			.name = "MESH_W", .base = mmMESH_W_PLL_BASE,
-	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "SRAM_W", .base = mmSRAM_W_PLL_BASE,
-	//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, -1, -1, -1 }
-		}, {
-			.name = "MESH_E", .base = mmMESH_E_PLL_BASE,
-	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "SRAM_E", .base = mmSRAM_E_PLL_BASE,
-	//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, -1, -1, -1 }
-		}, {
-			.name = "MME", .base = mmPSOC_MME_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
-	//		.nr = 0, .nf = 37, .od = 1, .nb = 18,
-			.nr = 4, .nf = 331, .od = 3, .nb = 165, //DAY 1 (830MHz)
-			.div_factor = { 0, 0, 4, 9 }
-		}, {
-			.name = "TPC", .base = mmPSOC_TPC_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "HBM", .base = mmPSOC_HBM_PLL_BASE,
-			.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
-			.div_factor = { 0, 1, -1, -1 }
-		}, {
-			.name = "NIC0", .base = mmNIC0_PLL_BASE,
-			.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
-			.div_factor = { 0, 4, 8, -1 }
-		}, {
-			.name = "NIC1", .base = mmNIC1_PLL_BASE,
-			.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
-			.div_factor = { 0, 4, 8, -1 }
-		}, {
-			.name = "NIC2_HBM", .base = mmNIC2_HBM_PLL_BASE,
-			.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 MHz */
-			.div_factor = { 0, 1, -1, -1 }
-		}, {
-			.name = "NIC2_MME", .base = mmNIC2_MME_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		}, {
-			.name = "NIC2_TPC", .base = mmNIC2_TPC_PLL_BASE,
-	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
-			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
-			.div_factor = { 0, 1, 2, 7 }
-		},
-};
-
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	int pll_num = ARRAY_SIZE(gaudi_pll_cfg);
-	int boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
-
-	if (boot_strap & SKIP_PLL_INIT)
-		return 0;
-
-	*hl_pll_cfg = gaudi_pll_cfg;
-
-	return pll_num;
-}
-
-void hl_uart_init(void)
-{
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
-
-	preboot_uart_init(pcie_div2_freq / HL_UART_BAUDRATE);
-}
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor;
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
-
-	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
-
-	return factor > 2 ? factor : 2;
-}
-
-int hl_pll_partial_init_get(void)
-{
-	return regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
-}
-
-void hl_pll_partial_init_set(void)
-{
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
-}
diff --git a/gaudi/gaudi.h b/gaudi/gaudi.h
deleted file mode 100644
index 0874123..0000000
--- a/gaudi/gaudi.h
+++ /dev/null
@@ -1,28 +0,0 @@
-/*
- * (C) Copyright 2019 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-/* Use DumpSel[9] to skip PLL initialization */
-#define SKIP_PLL_INIT BIT(17)
-
-#define mmDMA_W_PLL_BASE		0x487000
-#define mmIF_W_PLL_BASE			0x488000
-#define mmIF_E_PLL_BASE			0x4A7000
-#define mmDMA_E_PLL_BASE		0x4A8000
-#define mmMESH_W_PLL_BASE		0x4C7000
-#define mmSRAM_W_PLL_BASE		0x4C8000
-#define mmMESH_E_PLL_BASE		0x4E7000
-#define mmSRAM_E_PLL_BASE		0x4E8000
-#define mmPSOC_CPU_PLL_BASE		0xC70000
-#define mmPSOC_MME_PLL_BASE		0xC71000
-#define mmPSOC_PCI_PLL_BASE		0xC72000
-#define mmPSOC_TPC_PLL_BASE		0xC73000
-#define mmPSOC_HBM_PLL_BASE		0xC74000
-#define mmNIC0_PLL_BASE			0xCF9000
-#define mmNIC1_PLL_BASE			0xD39000
-#define mmNIC2_HBM_PLL_BASE		0xD78000
-#define mmNIC2_MME_PLL_BASE		0xD79000
-#define mmNIC2_TPC_PLL_BASE		0xD7A000
diff --git a/goya/common.c b/goya/common.c
new file mode 100644
index 0000000..d8a76a3
--- /dev/null
+++ b/goya/common.c
@@ -0,0 +1,237 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include <pcie_phy_regs.h>
+#include <psoc_global_conf_regs.h>
+#include <pcie_aux_regs.h>
+#include <pcie_core_regs.h>
+#include <cpu_timestamp_regs.h>
+#include <pcie_db_rsv_regs.h>
+
+#include <preboot.h>
+#include <pll.h>
+#include <uart.h>
+
+#include "common.h"
+
+#define BTL_COUNTER_MAX		BIT(22)
+
+static int preboot_wait_for_bit(unsigned long reg, unsigned long timeout_ns)
+{
+	unsigned int val;
+	unsigned long count = (timeout_ns * SYS_REF_CLOCK) / 1000000000UL;
+
+	while (count--) {
+		val = regmap_readl(reg);
+		if (val & 1)
+			return 0;
+	}
+
+	return 1;
+}
+
+static void preboot_wait_for_ext_ld_done(void)
+{
+	/*
+	 * Wait for HW FSM to complete 16KB copy to PHY
+	 * this is enough to set ext_ld_done.
+	 * FSM will set ext_ld_done after 32KB but why wait
+	 * if we can already move on
+	 */
+	while ((regmap_readl(mmPCIE_PHY_CR_PARA_CMD) & 0xFFFF) <= 0xD780)
+		;
+	regmap_writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_0);
+	regmap_writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_1);
+	regmap_writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_2);
+	regmap_writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_3);
+}
+
+static void preboot_wait_for_stable_pclk(void)
+{
+	int ret;
+
+	while (1) {
+		ret = 0;
+		/* Wait for PCLK stable - check MPLLA [no timeout] */
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_0, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_1, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_2, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_3, 100);
+		if (!ret)
+			break;
+
+		ret = 0;
+		/* Wait for PCLK stable - check MPLLB [no timeout] */
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_0, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_1, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_2, 100);
+		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_3, 100);
+		if (!ret)
+			break;
+	}
+}
+
+/* Handle PCIe erratum: make sure reset_n de-asserted only after perst */
+void preboot_pcie_wait_for_link(void)
+{
+	preboot_wait_for_ext_ld_done();
+
+	/* Wait for PRSTn */
+	while (!regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN))
+		;
+
+	/* Enable FW FSM after PRSTn */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PCI_FW_FSM);
+
+	preboot_wait_for_stable_pclk();
+
+	while (1) {
+		unsigned int prstn, ltssm, smlh, rdlh, hot_reset;
+
+		prstn = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN);
+		ltssm = regmap_readl(mmPCIE_AUX_LTSSM_EN);
+		smlh = regmap_readl(mmPCIE_AUX_SMLH_LINK_UP);
+		rdlh = regmap_readl(mmPCIE_AUX_RDLH_LINK_UP);
+		hot_reset = regmap_readl(mmPCIE_CORE_HOT_RST);
+
+		if ((hot_reset & BIT(0)) == BIT(0)) {
+			regmap_writel(0x10, mmPCIE_CORE_HOT_RST);
+			regmap_writel(1, mmPCIE_AUX_LTSSM_EN);
+		}
+
+		if (prstn && smlh && rdlh)
+			break;
+
+		/*
+		 * Some host assert prstn during link up while moving to GEN3
+		 * This cause BAR size to be reset
+		 * We stop link up process by disabling ltssm
+		 * till BAR is reconfigured
+		 */
+		if (!prstn && ltssm) {
+			regmap_writel(0, mmPCIE_AUX_LTSSM_EN);
+			continue;
+		}
+
+		if (prstn && !ltssm) {
+			regmap_writel(1, mmPCIE_AUX_LTSSM_EN);
+			continue;
+		}
+	}
+}
+
+/* Handle BTL erratum: exhaust timeout counter before any SPI access */
+void preboot_delay(void)
+{
+	unsigned long cntpct, end;
+	int sysclk = SYS_REF_CLOCK, i;
+
+	/* timer init */
+	regmap_writel(sysclk, mmCPU_TIMESTAMP_CNTFID0);
+	regmap_writel(0x3, mmCPU_TIMESTAMP_CNTCR); /* EN | HDBG */
+
+	/* get begin counter value and calc end counter value */
+	asm volatile("mrs %0, cntpct_el0" : "=r" (cntpct));
+	end = cntpct + BTL_COUNTER_MAX;
+
+	for (i = 0; i < BTL_COUNTER_MAX; i++) {
+		asm volatile("mrs %0, cntpct_el0" : "=r" (cntpct));
+		if (cntpct >= end)
+			break;
+	}
+}
+
+void preboot_putc_virt_uart(const char c)
+{
+	/*
+	 * update vUART to HOST
+	 * DATA: mmPCIE_DB_RSV_RSV_0
+	 * STATUS: mmPCIE_DB_RSV_RSV_1
+	 */
+	regmap_writel(c, mmPCIE_DB_RSV_RSV_0);
+	regmap_writel(1, mmPCIE_DB_RSV_RSV_1);
+}
+
+u64 get_cs_pad_sel(void)
+{
+	return mmPSOC_GLOBAL_CONF_PAD_SEL_53;
+}
+
+u64 get_cpu_boot_status_reg(void)
+{
+	return mmPSOC_GLOBAL_CONF_WARM_REBOOT;
+}
+
+u64 get_upd_pending_sts_reg(void)
+{
+	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
+}
+
+u32 hl_skip_pcie_init(void)
+{
+	/* Use DumpSel[8] to skip PCIe initialization */
+	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(21);
+}
+
+u64 hl_pll_get_cpu_pll_base_off(void)
+{
+	return mmCPU_PLL_BASE;
+}
+
+u64 hl_pll_get_cpu_cycles_per_usec(void)
+{
+	return 1330000000ULL / USEC_PER_SEC;
+}
+
+static struct hl_pll_cfg goya_pll_cfg[] = {
+	{
+		.name = "CPU", .base = mmCPU_PLL_BASE,
+		.nr = 2, .nf = 159, .od = 1, .nb = 79,
+		.div_factor = { 0, -1, 1, -1 }
+	}, {
+		.name = "IC", .base = mmIC_PLL_BASE,
+		.nr = 0, .nf = 63, .od = 15, .nb = 31,
+		.div_factor = { 0, 1, 1, 2 }
+	}, {
+		.name = "MME", .base = mmPSOC_MME_PLL_BASE,
+		.nr = 0, .nf = 63, .od = 15, .nb = 31,
+		.div_factor = { 0, 1, 2, 8 }
+	},
+};
+
+int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
+{
+	int pll_num = ARRAY_SIZE(goya_pll_cfg);
+
+	*hl_pll_cfg = goya_pll_cfg;
+
+	return pll_num;
+}
+
+void hl_uart_init(void)
+{
+	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+
+	preboot_uart_init(pcie_div2_freq / HL_UART_BAUDRATE);
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
+
+	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
+
+	return factor > 2 ? factor : 2;
+}
+
+int hl_pll_partial_init_get(void)
+{
+	return 0;
+}
+
+void hl_pll_partial_init_set(void) {}
diff --git a/goya/common.h b/goya/common.h
new file mode 100644
index 0000000..4e2256e
--- /dev/null
+++ b/goya/common.h
@@ -0,0 +1,11 @@
+/*
+ * (C) Copyright 2019 HabanaLabs, Ltd.
+ * Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#define mmCPU_PLL_BASE		0x4A2000
+#define mmIC_PLL_BASE		0x4A3000
+#define mmPSOC_MME_PLL_BASE	0xC71000
+#define mmPSOC_PCI_PLL_BASE	0xC72000
diff --git a/goya/goya.c b/goya/goya.c
deleted file mode 100644
index 6633585..0000000
--- a/goya/goya.c
+++ /dev/null
@@ -1,237 +0,0 @@
-/*
- * (C) Copyright 2019 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-#include <pcie_phy_regs.h>
-#include <psoc_global_conf_regs.h>
-#include <pcie_aux_regs.h>
-#include <pcie_core_regs.h>
-#include <cpu_timestamp_regs.h>
-#include <pcie_db_rsv_regs.h>
-
-#include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-
-#include "goya.h"
-
-#define BTL_COUNTER_MAX		BIT(22)
-
-static int preboot_wait_for_bit(unsigned long reg, unsigned long timeout_ns)
-{
-	unsigned int val;
-	unsigned long count = (timeout_ns * SYS_REF_CLOCK) / 1000000000UL;
-
-	while (count--) {
-		val = regmap_readl(reg);
-		if (val & 1)
-			return 0;
-	}
-
-	return 1;
-}
-
-static void preboot_wait_for_ext_ld_done(void)
-{
-	/*
-	 * Wait for HW FSM to complete 16KB copy to PHY
-	 * this is enough to set ext_ld_done.
-	 * FSM will set ext_ld_done after 32KB but why wait
-	 * if we can already move on
-	 */
-	while ((regmap_readl(mmPCIE_PHY_CR_PARA_CMD) & 0xFFFF) <= 0xD780)
-		;
-	regmap_writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_0);
-	regmap_writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_1);
-	regmap_writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_2);
-	regmap_writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_3);
-}
-
-static void preboot_wait_for_stable_pclk(void)
-{
-	int ret;
-
-	while (1) {
-		ret = 0;
-		/* Wait for PCLK stable - check MPLLA [no timeout] */
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_0, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_1, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_2, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLA_STATE_3, 100);
-		if (!ret)
-			break;
-
-		ret = 0;
-		/* Wait for PCLK stable - check MPLLB [no timeout] */
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_0, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_1, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_2, 100);
-		ret |= preboot_wait_for_bit(mmPCIE_PHY_PHYX_MPLLB_STATE_3, 100);
-		if (!ret)
-			break;
-	}
-}
-
-/* Handle PCIe erratum: make sure reset_n de-asserted only after perst */
-void preboot_pcie_wait_for_link(void)
-{
-	preboot_wait_for_ext_ld_done();
-
-	/* Wait for PRSTn */
-	while (!regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN))
-		;
-
-	/* Enable FW FSM after PRSTn */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PCI_FW_FSM);
-
-	preboot_wait_for_stable_pclk();
-
-	while (1) {
-		unsigned int prstn, ltssm, smlh, rdlh, hot_reset;
-
-		prstn = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN);
-		ltssm = regmap_readl(mmPCIE_AUX_LTSSM_EN);
-		smlh = regmap_readl(mmPCIE_AUX_SMLH_LINK_UP);
-		rdlh = regmap_readl(mmPCIE_AUX_RDLH_LINK_UP);
-		hot_reset = regmap_readl(mmPCIE_CORE_HOT_RST);
-
-		if ((hot_reset & BIT(0)) == BIT(0)) {
-			regmap_writel(0x10, mmPCIE_CORE_HOT_RST);
-			regmap_writel(1, mmPCIE_AUX_LTSSM_EN);
-		}
-
-		if (prstn && smlh && rdlh)
-			break;
-
-		/*
-		 * Some host assert prstn during link up while moving to GEN3
-		 * This cause BAR size to be reset
-		 * We stop link up process by disabling ltssm
-		 * till BAR is reconfigured
-		 */
-		if (!prstn && ltssm) {
-			regmap_writel(0, mmPCIE_AUX_LTSSM_EN);
-			continue;
-		}
-
-		if (prstn && !ltssm) {
-			regmap_writel(1, mmPCIE_AUX_LTSSM_EN);
-			continue;
-		}
-	}
-}
-
-/* Handle BTL erratum: exhaust timeout counter before any SPI access */
-void preboot_delay(void)
-{
-	unsigned long cntpct, end;
-	int sysclk = SYS_REF_CLOCK, i;
-
-	/* timer init */
-	regmap_writel(sysclk, mmCPU_TIMESTAMP_CNTFID0);
-	regmap_writel(0x3, mmCPU_TIMESTAMP_CNTCR); /* EN | HDBG */
-
-	/* get begin counter value and calc end counter value */
-	asm volatile("mrs %0, cntpct_el0" : "=r" (cntpct));
-	end = cntpct + BTL_COUNTER_MAX;
-
-	for (i = 0; i < BTL_COUNTER_MAX; i++) {
-		asm volatile("mrs %0, cntpct_el0" : "=r" (cntpct));
-		if (cntpct >= end)
-			break;
-	}
-}
-
-void preboot_putc_virt_uart(const char c)
-{
-	/*
-	 * update vUART to HOST
-	 * DATA: mmPCIE_DB_RSV_RSV_0
-	 * STATUS: mmPCIE_DB_RSV_RSV_1
-	 */
-	regmap_writel(c, mmPCIE_DB_RSV_RSV_0);
-	regmap_writel(1, mmPCIE_DB_RSV_RSV_1);
-}
-
-u64 get_cs_pad_sel(void)
-{
-	return mmPSOC_GLOBAL_CONF_PAD_SEL_53;
-}
-
-u64 get_cpu_boot_status_reg(void)
-{
-	return mmPSOC_GLOBAL_CONF_WARM_REBOOT;
-}
-
-u64 get_upd_pending_sts_reg(void)
-{
-	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
-}
-
-u32 hl_skip_pcie_init(void)
-{
-	/* Use DumpSel[8] to skip PCIe initialization */
-	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(21);
-}
-
-u64 hl_pll_get_cpu_pll_base_off(void)
-{
-	return mmCPU_PLL_BASE;
-}
-
-u64 hl_pll_get_cpu_cycles_per_usec(void)
-{
-	return 1330000000ULL / USEC_PER_SEC;
-}
-
-static struct hl_pll_cfg goya_pll_cfg[] = {
-	{
-		.name = "CPU", .base = mmCPU_PLL_BASE,
-		.nr = 2, .nf = 159, .od = 1, .nb = 79,
-		.div_factor = { 0, -1, 1, -1 }
-	}, {
-		.name = "IC", .base = mmIC_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 15, .nb = 31,
-		.div_factor = { 0, 1, 1, 2 }
-	}, {
-		.name = "MME", .base = mmPSOC_MME_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 15, .nb = 31,
-		.div_factor = { 0, 1, 2, 8 }
-	},
-};
-
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	int pll_num = ARRAY_SIZE(goya_pll_cfg);
-
-	*hl_pll_cfg = goya_pll_cfg;
-
-	return pll_num;
-}
-
-void hl_uart_init(void)
-{
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
-
-	preboot_uart_init(pcie_div2_freq / HL_UART_BAUDRATE);
-}
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor;
-	u32 pcie_div2_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV2);
-
-	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
-
-	return factor > 2 ? factor : 2;
-}
-
-int hl_pll_partial_init_get(void)
-{
-	return 0;
-}
-
-void hl_pll_partial_init_set(void) {}
diff --git a/goya/goya.h b/goya/goya.h
deleted file mode 100644
index 4e2256e..0000000
--- a/goya/goya.h
+++ /dev/null
@@ -1,11 +0,0 @@
-/*
- * (C) Copyright 2019 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-#define mmCPU_PLL_BASE		0x4A2000
-#define mmIC_PLL_BASE		0x4A3000
-#define mmPSOC_MME_PLL_BASE	0xC71000
-#define mmPSOC_PCI_PLL_BASE	0xC72000
